src/realm/views.cpp

```diff
@@ -215,38 +362,56 @@ void RowIndexes::do_sort(const SortDescriptor& order, const SortDescriptor& dist
             ++detached_ref_count;
     }
 
-    if (distinct) {
-        auto sorting_predicate = distinct.sorter(m_row_indexes);
-
-        // Remove all rows which have a null link along the way to the distinct columns
-        if (sorting_predicate.has_links()) {
-            v.erase(std::remove_if(v.begin(), v.end(),
-                                   [&](auto&& index) { return sorting_predicate.any_is_null(index); }),
-                    v.end());
+    const int num_descriptors = int(ordering.size());
+    for (int desc_ndx = 0; desc_ndx < num_descriptors; ++desc_ndx) {
+        const CommonDescriptor* common_descr = ordering[desc_ndx];
+        const SortDescriptor* sort_descr = dynamic_cast<const SortDescriptor*>(common_descr);
+        if (sort_descr) {
```

Comment:
Moving the declaration of `sort_descr` into the `if` would limit its scope to the body of the `if`, which is the only place where it can be usefully used.